featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -517,6 +517,50 @@ def month(vals):
         return month
 
 
+class PartOfDay(TransformPrimitive):
+    """Determines the part of day of a datetime.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> dates = [datetime(2020, 1, 11, 6, 2, 1),
+        ...          datetime(2021, 3, 31, 4, 2, 1),
+        ...          datetime(2020, 3, 4, 9, 2, 1)]
+        >>> part_of_day = PartOfDay()
+        >>> part_of_day(dates).tolist()
+        >>> ["early morning", "dawn", "late morning"]
+    """
+
+    name = "part_of_day"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(logical_type=Categorical, semantic_tags={"category"})
+    compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
+    description_template = "the part of day {} falls in"
+
+    def get_part_of_day(self, hour):
+        if hour in [4, 5]:
+            return "dawn"
+        elif hour in [6, 7]:
+            return "early morning"
+        elif hour in [8, 9, 10]:
+            return "late morning"
+        elif hour in [11, 12, 13]:
+            return "noon"
+        elif hour in [14, 15, 16]:
+            return "afternoon"
+        elif hour in [17, 18, 19]:
+            return "evening"
+        elif hour in [20, 21, 22]:
+            return "night"
+        elif hour in [23, 24, 1, 2, 3]:
+            return "midnight"
+
+    def get_function(self):
+        def part_of_day(vals):
+            return vals.apply(lambda x: self.get_part_of_day(x.hour))
```

Comment:
Can we see if there is a faster way to implement this primitive? Can we vectorize it or look for a faster approach? Perhaps a pd.merge?

Reply:
It also might be worth doing performance checking and using the fastest approach

Reply:
Sounds good, I will look into it right now

Reply:
I'm still trying to get it to work -- haven't been able to find a way that doesn't involve a map of some kind (because the output is not boolean). Does anyone have any tips? Right now it's about 4x slower than something like IsWeekend().